lib/Basics/Exceptions.cpp

```diff
@@ -170,3 +154,17 @@ std::string Exception::FillFormatExceptionString(char const* format, ...) {
       << "Failed to create an error message, giving up. " << errorMessage;
   FATAL_ERROR_EXIT();
 }
+
+[[noreturn]] void ::arangodb::basics::abortOrThrowException(Exception e) {
+#ifndef ARANGODB_ENABLE_MAINTAINER_MODE
+  throw e;
+#else
+  try {
+    throw std::move(e);
+  } catch (::arangodb::basics::Exception const& ex) {
+    LOG_TOPIC("fa7a1", FATAL, ::arangodb::Logger::CRASH) << ex.message();
+    TRI_ASSERT(false);
+  }
+  TRI_ASSERT(false);
```

Comment:
What is the added value of throwing the exception and catching it immediately after?
Can't we just log its message directly and then abort?

Reply:
True, thanks for noticing, that's a relict of incomplete refactoring. Will fix it.